server/src/test/java/org/elasticsearch/search/SearchServiceTests.java

```diff
@@ -1946,6 +1944,63 @@ public void testEnableSearchWorkerThreads() throws IOException {
         }
     }
 
+    public void testEnableQueryPhaseParallelCollection() throws IOException {
+        IndexService indexService = createIndex("index", Settings.EMPTY);
+        IndexShard indexShard = indexService.getShard(0);
+        ShardSearchRequest request = new ShardSearchRequest(
+            OriginalIndices.NONE,
+            new SearchRequest().allowPartialSearchResults(randomBoolean()),
+            indexShard.shardId(),
+            0,
+            indexService.numberOfShards(),
+            AliasFilter.EMPTY,
+            1f,
+            System.currentTimeMillis(),
+            null
+        );
+
+        ThreadPoolExecutor mockThreadPoolExecutor = mock();
+        int executorPoolSize = randomIntBetween(1, 100);
+        when(mockThreadPoolExecutor.getMaximumPoolSize()).thenReturn(executorPoolSize);
+
+        ExecutorService mockNotTPExecutor = mock();
```

Comment:
I have a slight preference to avoid mocking, just because it makes things very artificial and not using mocks pushes us to make things more testable without using mocks :)  Could we use real executors here?